### PR TITLE
AG-5988 - More parameters for pie label formatter

### PR DESCRIPTION
**Pull request by alexanderby**
https://ag-grid.atlassian.net/browse/AG-5988

Improvements for Pie/Doughnut Chart segment label formatter:
- Allow access to associated data record through `datum` property.
- Access to config properties `labelKey`, `labelName`, `labelValue`, `angleKey`, `angleName`, `angleValue`, `radiusKey`, `radiusName`, `radiusValue`.
- `value` property becomes deprecated.

### CODE REVIEW

File: community-modules/core/src/ts/interfaces/iAgChartOptions.ts [hunk 1]
```diff
@@ -1239,6 +1239,39 @@ export interface AgPieSeriesTooltipRendererParams extends AgPolarSeriesTooltipRe
     labelName?: string;
 }
 
+export interface AgPieSeriesLabelFormatterParams<DatumType> {
+    /** Datum from the series data array that the tooltip is being rendered for. */
```
Comment (manapeirov) [inline community-modules/core/src/ts/interfaces/iAgChartOptions.ts:1243]: *label

**Review by manapeirov — COMMENTED** [review]
Looks good Alex

**Comment by manapeirov** [comment]
I ran the tests and they seem fine

File: grid-packages/ag-grid-docs/documentation/doc-pages/charts-pie-series/index.md [hunk 2]
```diff
@@ -51,6 +51,22 @@ series: [{
 }]
 ```
 
+The labels can be customised via the `label.formatter` function.
+The `datum` property provides access to a data item associated with a segment.
```
Comment (manapeirov) [inline grid-packages/ag-grid-docs/documentation/doc-pages/charts-pie-series/index.md:55]: The formatter only allows changing the content or text displayed as the label, rather than full customisation. Maybe it would be better to rephrase this to something along the lines of: 

A label formatter function can be used to change the value displayed in the label. The `label.formatter` function receives a single params object containing values associated with a pie segment. please see the [API reference](#api-reference) for the full list of provided properties.

For example, to display the angle values for a given pie sector in the label, you can use the following label formatter function:

```js
series: [{
    ...
    label: {
        formatter: ({ angleValue }) => {
            return `${angleValue}`;
        }
    }
}]
```

Comment (alexanderby) [reply grid-packages/ag-grid-docs/documentation/doc-pages/charts-pie-series/index.md:55]: Sure. One problem I see is that some people may confuse `angleValue` with angle value, e.g. 45° or 0.5 radians.

Comment (manapeirov) [reply grid-packages/ag-grid-docs/documentation/doc-pages/charts-pie-series/index.md:55]: Good point! Feel free to modify the snippet as you see appropriate

**Review by manapeirov — APPROVED** [review]
LGTM! :)
